aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -0,0 +1,978 @@
+        let principal = coin::value(&coins);
+        assert!(principal >= min_stake_required, error::invalid_argument(EINSUFFICIENT_STAKE_AMOUNT));
+
+        // Initialize Store resource if this is the first time the staker has delegated to anyone.
+        let staker_address = signer::address_of(staker);
+        if (!exists<Store>(staker_address)) {
+            move_to(staker, new_staking_contracts_holder(staker));
+        };
+
+        // Cannot create the staking contract if it already exists.
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contracts = &mut store.staking_contracts;
+        assert!(
+            !simple_map::contains_key(staking_contracts, &operator),
+            error::invalid_argument(ESTAKING_CONTRACT_ALREADY_EXISTS)
+        );
+
+        // Initialize the stake pool in a new resource account. This allows the same staker to contract with multiple
+        // different operators.
+        let (stake_pool_signer, stake_pool_signer_cap, owner_cap) =
+            create_stake_pool(staker, operator, voter, contract_creation_seed);
+
+        // Add the stake to the stake pool.
+        stake::add_stake_with_cap(&owner_cap, coins);
+
+        // Create the contract record.
+        let pool_address = signer::address_of(&stake_pool_signer);
+        simple_map::add(staking_contracts, operator, StakingContract {
+            principal,
+            pool_address,
+            owner_cap,
+            commission_percentage,
+            // Make sure we don't have too many pending recipients in the distribution pool.
+            // Otherwise, a griefing attack is possible where the staker can keep switching operators and create too
+            // many pending distributions. This can lead to out-of-gas failure whenever distribute() is called.
+            distribution_pool: pool_u64::create(MAXIMUM_PENDING_DISTRIBUTIONS),
+            signer_cap: stake_pool_signer_cap,
+        });
+
+        emit_event(
+            &mut store.create_staking_contract_events,
+            CreateStakingContractEvent { operator, voter, pool_address, principal, commission_percentage },
+        );
+        pool_address
+    }
+
+    /// Add more stake to an existing staking contract.
+    public entry fun add_stake(staker: &signer, operator: address, amount: u64) acquires Store {
+        let staker_address = signer::address_of(staker);
+        assert_staking_contract_exists(staker_address, operator);
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+
+        // Add the stake to the stake pool.
+        let staked_coins = coin::withdraw<AptosCoin>(staker, amount);
+        stake::add_stake_with_cap(&staking_contract.owner_cap, staked_coins);
+
+        staking_contract.principal = staking_contract.principal + amount;
```

Comment:
why aren't we increasing the pools total coins?

Reply:
The pool is for distribution only. This function is adding stake, not taking anything out so the distribution pool should not be updated.